src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -71,31 +76,43 @@
                           (.printStackTrace exception out))
                         attempts))))
 
+(defn- strip-metadata-suffix
+  [queue-metadata-str]
+  (str/replace queue-metadata-str #"\.json$" ""))
+
 (defn- entry-cmd-data-filename
   [entry]
-  ;; 10291-1469469689-cat-4-foo.org.json
+  ;; 10291_1469469689_cat_4_foo.org.json
   (let [id (stock/entry-id entry)
-        meta (stock/entry-meta entry)]
-    (str id \- meta)))
+        metadata (stock/entry-meta entry)]
+    (str id \_
+         (utf8-truncate metadata (- 255
```

Comment:
Not sure how yet, but feel like we might want to abstract this a bit more, i.e. so that everyone just calls (constrain-metadata ...) or similar, and it worries about the details.